File: core/src/main/java/info/nightscout/androidaps/plugins/pump/common/defs/PumpHistoryEntryGroup.java
package info.nightscout.androidaps.plugins.pump.common.defs;

import java.util.ArrayList;
import java.util.List;


import info.nightscout.androidaps.core.R;
import info.nightscout.androidaps.utils.resources.ResourceHelper;


/**
 * This file was taken from GGC - GNU Gluco Control (ggc.sourceforge.net), application for diabetes
 * management and modified/extended for AAPS.
 * <p>
 * Author: Andy {andy.rozman@gmail.com}
 */

public enum PumpHistoryEntryGroup {

    All(R.string.history_group_all),
    Bolus(R.string.history_group_bolus),
    Basal(R.string.history_group_basal),
    Prime(R.string.history_group_prime),
    Configuration(R.string.history_group_configuration),
    Alarm(R.string.history_group_alarm),
    Glucose(R.string.history_group_glucose),
    Notification(R.string.history_group_notification),
    Statistic(R.string.history_group_statistic),
    Unknown(R.string.history_group_unknown),
    ;

    private final int resourceId;
    private String translated;

    private static List<PumpHistoryEntryGroup> translatedList;

    PumpHistoryEntryGroup(int resourceId) {
        this.resourceId = resourceId;
    }

    private static void doTranslation(ResourceHelper resourceHelper) {
        translatedList = new ArrayList<>();

        for (PumpHistoryEntryGroup pumpHistoryEntryGroup : values()) {
            pumpHistoryEntryGroup.translated = resourceHelper.gs(pumpHistoryEntryGroup.resourceId);
            translatedList.add(pumpHistoryEntryGroup);
        }
    }

    public static List<PumpHistoryEntryGroup> getTranslatedList(ResourceHelper resourceHelper) {
        if (translatedList == null) doTranslation(resourceHelper);
        return translatedList;
    }

    public int getResourceId() {
        return resourceId;
    }

    public String getTranslated() {
        return translated;
    }

    public String toString() {
        return this.translated;
    }
}

File: pump-common/src/main/java/info/nightscout/androidaps/plugins/pump/common/defs/PumpHistoryEntryGroup.kt
package info.nightscout.androidaps.plugins.pump.common.defs

import info.nightscout.androidaps.plugins.pump.common.R
import info.nightscout.androidaps.utils.resources.ResourceHelper
import java.util.*

/**
 * This file was taken from GGC - GNU Gluco Control (ggc.sourceforge.net), application for diabetes
 * management and modified/extended for AAPS.
 *
 *
 * Author: Andy {andy.rozman@gmail.com}
 */
enum class PumpHistoryEntryGroup(val resourceId: Int) {

    All(R.string.history_group_all),
    Bolus(R.string.history_group_bolus),
    Basal(R.string.history_group_basal),
    Prime(R.string.history_group_prime),
    Configuration(R.string.history_group_configuration),
    Alarm(R.string.history_group_alarm),
    Glucose(R.string.history_group_glucose),
    Notification(R.string.history_group_notification),
    Statistic(R.string.history_group_statistic),
    Unknown(R.string.history_group_unknown);

    var translated: String? = null
        private set

    override fun toString(): String {
        return translated!!
    }

    companion object {

        private var translatedList: MutableList<PumpHistoryEntryGroup>? = null

        private fun doTranslation(rh: ResourceHelper) {
            translatedList = ArrayList()
            for (pumpHistoryEntryGroup in values()) {
                pumpHistoryEntryGroup.translated = rh.gs(pumpHistoryEntryGroup.resourceId)
                (translatedList as ArrayList<PumpHistoryEntryGroup>).add(pumpHistoryEntryGroup)
            }
        }

        fun getTranslatedList(rh: ResourceHelper): List<PumpHistoryEntryGroup> {
            if (translatedList == null) doTranslation(rh)
            return translatedList!!
        }
    }

}